GithubApp/Controllers/SearchViewController.swift

```diff
@@ -48,25 +72,86 @@ extension SearchViewController: UITableViewDataSource{
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "SearchTableViewCell", for: indexPath) as! SearchTableViewCell
+        let cell: SearchTableViewCell = tableView.dequeueCell(for: indexPath)
         let personData = searchedData[indexPath.row]
         cell.backgroungImage.image = UIImage(named: "R")
-        cell.personImage.loadFrom(URLAddress: personData.owner!.avatarUrl)
         cell.nameLabel.text = personData.name
         cell.descriptionLabel.text = personData.description
+        cell.languageLabel.text = personData.languageUrl.rawValue
+        guard let imageURL = personData.owner?.avatarUrl else {
+            cell.personImage.image = UIImage(named: "Image1")
+            return cell
+        }
+        cell.personImage.loadFrom(URLAddress: imageURL)
         return cell
     }
-    
 }
 
 extension SearchViewController: UISearchBarDelegate{
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
         searchedData.removeAll()
         for person in data {
-            if (person.name!.hasPrefix(searchText.lowercased())){
+            guard let personName = person.name else{
+                return
+            }
+            if (personName.hasPrefix(searchText.lowercased())){
                 searchedData.append(person)
             }
         }
         self.searchView.tableView.reloadData()
     }
 }
+
+extension SearchViewController: FilterDelegate{
+    
+    // MARK: Functions 
+    func getSelectedFilters(filters: [Filter]) {
+        self.filterData.removeAll()
+        self.filterData = filters
+        self.searchedData.removeAll()
+        var isFilterApplied :Bool = true
+        for filter in filters {
+            if(filter.isSelected){
+                isFilterApplied = false
+                break
+            }
+        }
```

Comment:
you can reduce this piece of code by using `contains` method of collections
```
let hasFilterSelected = filters.contains(where: { $0.isSelected })
```

Reply:
Changed it